lib/ext/adviser.rb

```diff
@@ -0,0 +1,19 @@
+require 'csv'
+
+module AdviserExtension
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def to_csv(_ = {})
+      CSV.generate do |csv|
+        csv << ['Ref. Number', 'Name', 'Firm']
+
+        find_each do |adviser|
+          csv << [adviser.reference_number, adviser.name, adviser.firm.registered_name]
+        end
+      end
+    end
+  end
+end
+
+Adviser.send(:include, AdviserExtension)
```

Comment:
@Tr4pSt3R @tomas-stefano I think looking at the current implementation of `.to_csv`, we've actually made it very custom to creating a specific csv file, which then constrains any further extension. IMO that perhaps means that we should not be extending the `.to_csv` method but renaming it as a new custom method.